third_party/intel/lib/TritonIntelGPUTransforms/AccelerateMatmul.cpp

```diff
@@ -126,6 +126,12 @@ class BlockedToDPAS : public OpRewritePattern<tt::DotOp> {
     unsigned repeatCount =
         std::min(dpasCap.repeatCount, (unsigned)retShape[rank - 2] /*M*/);
     unsigned threadsPerWarp = ttg::TritonGPUDialect::getThreadsPerWarp(mod);
+    unsigned numElemsPerRowForA =
+        opsPerChan == 1
+            ? dpasCap.systolicDepth
+            : dpasCap.systolicDepth * 2; // A is packed to i16 or i32.
```

Comment:
Wonder if below is correct...
```suggestion
    unsigned numElemsPerRowForA = dpasCap.systolicDepth * opsPerChan;
```

Reply:
For int8 case, we packed the int8 values to i16 for DPAS operation. There is only 16xi16  elements per row.

For the DPAS encoding, `repeatCount=1, systolicDepth =8, opsPerChan = 4, threadsPerWarp=32`, it is an invalid encoding. Because a single i16 scalar type in the SIMD-32 kernel represents 32xi16 elements. We cannot use the i16 type to refer a 16xi16 non-uniform value in SIMD-32 kernel. 
We have to use the `repeatCount=2, systolicDepth =8, opsPerChan = 4, threadsPerWarp=32` as the minimal DPAS encoding. For the M=1 block shape, the second row contains the duplicated value as the first row.

The expression `unsigned numElemsPerRowForA = dpasCap.systolicDepth * opsPerChan;` is not correct for int8 and threadsPerWarp=32 case.